src/helpers/subscribe.ts

```diff
@@ -0,0 +1,208 @@
+// TODO: need to see if EventEmitter works in the browser
+import { EventEmitter } from "node:events";
+import { type TablelandTables } from "@tableland/evm";
+import { pollTransactionReceipt } from "../validator/receipt.js";
+import {
+  getTableIdentifier,
+  getContractAndOverrides,
+  type TableIdentifier,
+} from "../registry/contract.js";
+import { extractBaseUrl, type Config } from "../helpers/index.js";
+
+type ContractEventTableIdMap = Record<
+  string,
+  {
+    tableIdIndex: number;
+    emit: string;
+  }
+>;
+
+type ContractMap = Record<number, TablelandTables>;
+
+type ListenerMap = Record<
+  string,
+  {
+    chainId: number;
+    tableId: string;
+    emitter: EventEmitter;
+  }
+>;
+
+/**
+ * List of the Registry Contract events that will be emitted from the TableEventBus.
+ */
+// TODO: this is an awkward way to keep track of the index of the Events' tableId index
+const contractEvents: ContractEventTableIdMap = {
+  RunSQL: {
+    tableIdIndex: 2,
+    emit: "change",
+  },
+  TransferTable: {
+    tableIdIndex: 2,
+    emit: "transfer",
+  },
+  SetController: {
+    tableIdIndex: 0,
+    emit: "set-controller",
+  },
+};
+
+/**
+ * TableEventBus provides a way to listen for:
+ *  mutations, transfers, and changes to controller
+ */
+export class TableEventBus {
+  readonly config: Config;
+  readonly contracts: ContractMap;
+  readonly listeners: ListenerMap;
+
+  /**
+   * Create a TableEventBus instance with the specified connection configuration.
+   * @param config The connection configuration. This must include an ethersjs
+   * Signer. If passing the config from a pre-existing Database instance, it
+   * must have a non-null signer key defined.
+   */
+  constructor(config: Partial<Config> = {}) {
+    /* c8 ignore next 3 */
+    if (config.signer == null) {
+      throw new Error("missing signer information");
+    }
+
+    this.config = config as Config;
+    this.contracts = {};
+    this.listeners = {};
+  }
+
+  /**
+   * Start listening to the Registry Contract for events that are associated
+   * with a given table.
+   * There's only ever one "listener" for a table, but the emitter that
+   * Contract listener has can have as many event listeners as the environment
+   * supports.
+   * @param tableName The full name of table that you want to listen for
+   * changes to.
+   */
+  async addTableListener(tableName: string): Promise<EventEmitter> {
```

Comment:
Why `addTableListener` versus just `addListener`? Just because name clash with classic listener adding api?

Reply:
I like it.  Not sure what my original thinking was, but I'll update it.